django/db/models/fields/related.py

```diff
@@ -1375,6 +1376,27 @@ def _check_relationship_model(self, from_model=None, **kwargs):
 
         return errors
 
+    def _check_table_uniqueness(self, **kwargs):
+        if isinstance(self.remote_field.through, six.string_types):
+            return []
+        registered_tables = {
+            model._meta.db_table: model
+            for model in self.opts.apps.get_models(include_auto_created=True)
+            if model != self.remote_field.through
+        }
+        m2m_db_table = self.m2m_db_table()
+        if m2m_db_table in set(registered_tables.keys()):
+            msg = 'Field is using a table that has already been registered by \'%s.%s.%s\'.'
+            table = registered_tables[m2m_db_table]
+            return [
+                checks.Error(
+                    msg % (table._meta.app_label, table._meta.object_name, self.name),
```

Comment:
You can replace `table._meta.app_label` and `table._meta.object_name` by `table._meta.label`